pkg/worker/k8spodtemplating.go

```diff
@@ -27,20 +29,32 @@ var (
 	nugetPackageScript string
 )
 
-func getPodSpec(ctx context.Context, step wharfyml.Step) (v1.Pod, error) {
+func getStepPodSpec(ctx context.Context, step wharfyml.Step) (v1.Pod, error) {
 	annotations := map[string]string{
-		"wharf.iver.com/step": step.Name,
+		"wharf.iver.com/step-name": step.Name,
 	}
 	if stage, ok := contextStageName(ctx); ok {
-		annotations["wharf.iver.com/stage"] = stage
+		annotations["wharf.iver.com/stage-name"] = stage
 	}
 	pod := v1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
 			GenerateName: getPodGenerateName(step),
 			Annotations:  annotations,
 			Labels: map[string]string{
-				"wharf.iver.com/build": "true",
+				"app":                          "wharf-cmd-worker-step",
+				"app.kubernetes.io/name":       "wharf-cmd-worker-step",
+				"app.kubernetes.io/part-of":    "wharf",
+				"app.kubernetes.io/managed-by": "wharf-cmd-worker",
+				"app.kubernetes.io/created-by": "wharf-cmd-worker",
+
+				"wharf.iver.com/instance":   "prod",
+				"wharf.iver.com/build-ref":  "123",
+				"wharf.iver.com/project-id": "456",
+				"wharf.iver.com/stage-id":   "789",
+				"wharf.iver.com/step-id":    "789",
+				"wharf.iver.com/build":      "true",
```

Comment:
> There is also the label `wharf.iver.com/build=true`, should this be kept in?
>
>https://github.com/iver-wharf/wharf-cmd/issues/17#issuecomment-1110779331

Some of these are of use, but some are better lived as annotations, if at all.

These are some really excess data here. There won't be that many steps in a single build that we need the optimization of label selectors to find them.

I think we should move the following to annotations:

- :twisted_rightwards_arrows: wharf.iver.com/project-id
- :twisted_rightwards_arrows: wharf.iver.com/stage-id
- :twisted_rightwards_arrows: wharf.iver.com/step-id

And then remove the following, as users should query for `app` or `app.kubernetes.io/managed-by` instead to filter on all builds:

- :x: wharf.iver.com/build